File: examples/scripts/ci/generate-matrix.test.js
const { generateMatrix } = require('./generate-matrix');

test('generateMatrix creates correct steps', () => {
  const platforms = [
    { os: 'linux', arch: 'amd64', image: 'golang:1.26-alpine' }
  ];
  
  const steps = generateMatrix(platforms);
  
  expect(steps).toHaveLength(1);
  expect(steps[0].id).toBe('build-linux-amd64');
  expect(steps[0].env.GOOS).toBe('linux');
});

test('generateMatrix handles multiple platforms', () => {
  const platforms = [
    { os: 'linux', arch: 'amd64', image: 'golang:1.26-alpine' },
    { os: 'darwin', arch: 'arm64', image: 'golang:1.26-alpine' }
  ];
  
  const steps = generateMatrix(platforms);
  
  expect(steps).toHaveLength(2);
  expect(steps[1].id).toBe('build-darwin-arm64');
});

File: examples/scripts/ci/generate_matrix_test.go
package main

import (
	"testing"
)

func TestGenerateMatrix(t *testing.T) {
	platforms := []Platform{
		{"linux", "amd64", "golang:1.26-alpine"},
		{"darwin", "arm64", "golang:1.26-alpine"},
	}

	steps := GenerateMatrix(platforms)

	if len(steps) != 2 {
		t.Errorf("expected 2 steps, got %d", len(steps))
	}

	if steps[0].ID != "build-linux-amd64" {
		t.Errorf("expected first step ID build-linux-amd64, got %s", steps[0].ID)
	}

	if steps[1].Env["GOARCH"] != "arm64" {
		t.Errorf("expected second step GOARCH arm64, got %s", steps[1].Env["GOARCH"])
	}
}
